optuna/visualization/matplotlib/_pareto_front.py

```diff
@@ -0,0 +1,237 @@
+
+    .. seealso::
+        Please refer to :func:`optuna.visualization.plot_pareto_front` for an example.
+
+    Example:
+
+        The following code snippet shows how to plot the Pareto front of a study.
+
+        .. plot::
+
+            import optuna
+
+
+            def objective(trial):
+                x = trial.suggest_float("x", 0, 5)
+                y = trial.suggest_float("y", 0, 3)
+
+                v0 = 4 * x ** 2 + 4 * y ** 2
+                v1 = (x - 5) ** 2 + (y - 5) ** 2
+                return v0, v1
+
+
+            study = optuna.create_study(directions=["minimize", "minimize"])
+            study.optimize(objective, n_trials=50)
+
+            optuna.visualization.matplotlib.plot_pareto_front(study)
+
+    Args:
+        study:
+            A :class:`~optuna.study.Study` object whose trials are plotted for their objective
+            values.
+        target_names:
+            Objective name list used as the axis titles. If :obj:`None` is specified,
+            "Objective {objective_index}" is used instead.
+        include_dominated_trials:
+            A flag to include all dominated trial's objective values.
+        axis_order:
+            A list of indices indicating the axis order. If :obj:`None` is specified,
+            default order is used.
+
+    Returns:
+        A :class:`matplotlib.axes.Axes` object.
+
+    Raises:
+        :exc:`ValueError`:
+            If the number of objectives of ``study`` isn't 2 or 3.
+    """
+
+    _imports.check()
+
+    if len(study.directions) == 2:
+        return _get_pareto_front_2d(study, target_names, include_dominated_trials, axis_order)
+    elif len(study.directions) == 3:
+        return _get_pareto_front_3d(study, target_names, include_dominated_trials, axis_order)
+    else:
+        raise ValueError("`plot_pareto_front` function only supports 2 or 3 objective studies.")
+
+
+def _get_non_pareto_front_trials(
+    study: Study, pareto_trials: List[FrozenTrial]
+) -> List[FrozenTrial]:
+
+    non_pareto_trials = []
+    for trial in study.get_trials():
+        if trial.state == TrialState.COMPLETE and trial not in pareto_trials:
+            non_pareto_trials.append(trial)
+    return non_pareto_trials
+
+
+def _get_pareto_front_2d(
+    study: Study,
+    target_names: Optional[List[str]],
+    include_dominated_trials: bool = False,
+    axis_order: Optional[List[int]] = None,
+) -> "Axes":
+
+    # Set up the graph style.
+    plt.style.use("ggplot")  # Use ggplot style sheet for similar outputs to plotly.
+    _, ax = plt.subplots()
+    ax.set_title("Pareto-front Plot")
+    cmap = plt.get_cmap("tab10")  # Use tab10 colormap for similar outputs to plotly.
+
+    if target_names is None:
+        target_names = ["Objective 0", "Objective 1"]
+    elif len(target_names) != 2:
+        raise ValueError("The length of `target_names` is supposed to be 2.")
+
+    ax.set_xlabel(target_names[0])
+    ax.set_ylabel(target_names[1])
```

Comment:
When I changed the axis order, the labels are not shown as expected. I think we need to refer to the values of the `axis_order` list similarly to the [plotly-based implementation](https://github.com/optuna/optuna/blob/3b8a5fe71dcd511d73070a43394a9e70341b4db9/optuna/visualization/_pareto_front.py#L173-L174).

```python
ax.set_xlabel(target_names[axis_order[0]]
ax.set_ylabel(target_names[axis_order[1]]
```

`axis_order=[0, 1]` (default)
![image](https://user-images.githubusercontent.com/3255979/114194595-a18f6d00-998a-11eb-857a-4db0a73184ae.png)

`axis_order=[1, 0]`: The axis labels are not swapped.
![image](https://user-images.githubusercontent.com/3255979/114194635-abb16b80-998a-11eb-9d7a-6f2180e6e3da.png)